### DISCUSSION TITLE
SSG vs SSR for CRUD operations

### DISCUSSION THREAD

**Discussion by credod** [vercel/next.js#28473] [Help]
Hello! I’m new to SSG and SSR concepts. I’ve gone through the [tutorial from the Next.js website](https://nextjs.org/learn/basics/create-nextjs-app?utm_source=next-site&utm_medium=nav-cta&utm_campaign=next-website) and I think I’m starting to understand the concepts but I’m still a bit confused. If I was to take the same blog app from the [tutorial demo](https://next-learn-starter.vercel.app/) but added CRUD capabilities to the _post_ page (`pages/posts/[id].js`), would I need to change this page to be Server-side rendered instead of Statically generated? Would making use of ISR here be an ok choice?

I think I'm struggling to visualize real-life scenarios of when we would use each form of pre-rendering. If anyone could shed some light on this I would appreciate it! Thank you!

**Comment by oliverloops** [comment]
Hi @credod, at the beginning don't worry if you don't fully understand how every solution works from inside, this concepts  sometimes tend to be tricky to differentiate when you're starting. Now going forward with your question, In short the answers is: for small projects keep with SSG (more specifically for your blog website). But Let me explain you why.

In order to create this CRUD functionality you mention, is recommended to use SSG for a simple and small blog where you just display plain text and some images, for performance reasons is better to choose SSG because all happens at build time and you don't need to refresh your pages or content frequently. but if you are planing to scale to hundreds or thousands of pages it's fully recommended to go with ISR and that's because some of the built-in features with this approach, one of the most important is the pre-caching applied to some pages that don't need to load at build time but that eventually will be requested, you have the option to revalidate data too. SSR is not fully recommended and should be used as a last resource just or if you need real time updates from the server and if your data is changing all the time, this approach fits best for more niche solutions. 

 I'll hope this helps.

**AS A BONUS:**
If you're interested on learn more about ISR, I recommend to watch this YouTube video(https://www.youtube.com/watch?v=nrfuN_Hyd3Y) from @leerob, it's super well explained with a real-life example and is less than 10 minutes. 👌

**Comment by credod** [comment]
Thank you for your reply @oliverloops! Your explanation and the video are very helpful!
